client-v2/src/reducers/frontsReducer.ts

```diff
@@ -42,6 +42,13 @@ const reducer = (
     };
   }
   switch (action.type) {
+    case 'FETCH_UPDATE_METADATA_SUCCESS': {
+      return set(
+        ['frontsConfig', 'data', 'fronts', action.payload.frontId, 'metadata'],
+        action.payload.metadata,
+        newState
+      );
+    }
```

Comment:
Re: our conversation yesterday, this is fine but I understand why it might make you feel a little uneasy! To mitigate, we could always create a new reducer that extended the `frontsConfigReducer` and add our action there to keep our state management on the same 'level', e.g.

```
const newConfigReducer = (state, action) => {
  newState = frontsConfigReducer(state, action)
  switch (action.type) {
    case 'FETCH_UPDATE_METADATA_SUCCESS': {
       // ur logic here omg
    }
   // ...etc
}
```

... which should `just work` when eventually imported into the parent `frontsReducer`.

Reply:
Alrighty, if you're ok with the slight code smell then I'll create a client side integration test incase we change how bundles work and call it a day.